### PR TITLE
#77 login page

### LINKED ISSUE #77: User login/signup component

### PR DESCRIPTION
**Pull request by knox153**
Closes #77

Finished the login page UI 🎉

## Login Page
I added a error icon in the input field to indicate the corresponding error field. Also made the page more responsive.
image

### CODE REVIEW

File: packages/server/server/api/user/user.service.ts [hunk 1]
```diff
@@ -44,25 +44,35 @@ class UserService {
             });
     }
 
-    public checkDuplicateUser(userData: User): Promise<boolean> {
+    public checkDuplicateUser(userData: User): Promise<string | null> {
```
Comment (shizuko-akamoto) [inline packages/server/server/api/user/user.service.ts:47]: Maybe you could do something like this for readability.
```
const duplicateEmail = userModel.findOne({ email: userData.email })
  .collation({ locale: 'en', strength: 2 })
  .exec();
const duplicateUsername = userModel.findOne({ username: userData.username })
  .collation({ locale: 'en', strength: 2 }).exec();

return Promise.all([duplicateEmail, duplicateUsername]).then(([byEmail, byUsername]: [User, User]) => {
    if (byEmail) {
        return Promise.resolve('Email');
    } else if (byUsername) {
        return Promise.resolve('Username');
    } else {
        return Promise.resolve(null);
    }
})
.catch((err) => {
     logger.warn(`Failed to find user`);
      return Promise.reject(
            new InternalServerError({
                message: `Failed to find user: ${err.toString()}`,
            })
       );
});
```

File: packages/client/src/pages/Login/login.scss [hunk 2]
```diff
@@ -1,53 +1,167 @@
...
+                    margin-bottom: 0.625rem;
+                    text-transform: uppercase;
+                }
+
+                input {
+                    display: block;
+                    width: 100%;
+                    height: 2.5rem;
+                    padding: 0 2rem 0 0.625rem;
+                    background: rgba(255, 255, 255, 0);
```
Comment (shizuko-akamoto) [inline packages/client/src/pages/Login/login.scss:82]:
```suggestion
                    background: rgba($tc-white, 0);
```

File: packages/client/src/pages/Login/login.scss [hunk 3]
```diff
@@ -1,53 +1,167 @@
...
+                    color: $tc-black;
+                    border: 1px solid $silver-gray;
+                    border-radius: 0.5rem;
+
+                    &:focus {
+                        outline: none;
+                        background: rgba(255, 255, 255, 0.5);
```
Comment (shizuko-akamoto) [inline packages/client/src/pages/Login/login.scss:89]: Same here

File: packages/client/src/pages/Login/login.scss [hunk 4]
```diff
@@ -1,53 +1,167 @@
...
+                    width: 100%;
+                    height: 2.5rem;
+                    margin: 1.25rem 0 0.625rem 0;
+                    text-transform: uppercase;
+                    text-align: center;
+                    color: $tc-white;
+                    font-weight: bold;
+                    border: none;
+                    border-radius: 0.5rem;
+                    background-color: rgba(13, 59, 104, 1);
```
Comment (shizuko-akamoto) [inline packages/client/src/pages/Login/login.scss:107]: Does this colour correspond to any colour variable predefined? not sure lol

File: packages/server/server/api/user/user.service.ts (hunk 1 above)
Comment (knox153) [reply packages/server/server/api/user/user.service.ts:47]: Thank you!! I forgot `Promise.all` is a thing, and this is so much better!

File: packages/client/src/pages/Login/login.scss (hunk 2 above)
Comment (knox153) [reply packages/client/src/pages/Login/login.scss]: Dangg, I didn't realize scss can convert hex to rgba for us, that is so useful haha

File: packages/client/src/pages/Login/login.scss (hunk 4 above)
Comment (knox153) [reply packages/client/src/pages/Login/login.scss]: It is $indigo-dye 😛 Updated to use the corresponding variable!

**Pull request merged**